@here/harp-mapview/lib/Tile.ts

```diff
@@ -826,6 +660,41 @@ export class Tile implements CachedResource {
         return this.m_disposed;
     }
 
+    /**
+     * Gets the [[TileGeometryLoader]] that manages this tile.
+     */
+    get tileGeometryLoader(): TileGeometryLoader | undefined {
+        return this.m_tileGeometryLoader;
+    }
+
+    /**
+     * Sets the [[TileGeometryLoader]] to manage this tile.
+     *
+     * @param tileGeometryLoader A [[TileGeometryLoader]] instance to manage the geometry creation
+     *      for this tile.
+     */
+    set tileGeometryLoader(tileGeometryLoader: TileGeometryLoader | undefined) {
+        this.m_tileGeometryLoader = tileGeometryLoader;
+    }
+
+    /**
+     * `True` if the basic geometry has been loaded, and the `Tile` is ready  for display.
+     */
+    get basicGeometryLoaded(): boolean {
+        return this.m_tileGeometryLoader === undefined
+            ? this.hasGeometry
+            : this.m_tileGeometryLoader.basicGeometryLoaded || this.m_tileGeometryLoader.isFinished;
```

Comment:
Why are we checking isFinished here? Can "isFinished" be set and "basicGeometryLoaded" not be set? If so does that really mean the basicGeometry is loaded?

Reply:
It could happen that the tile loader did not provide any geometry, and now it is finished. Bummer, but we should catch that.